crates/wasmtime/src/trap.rs

```diff
@@ -31,10 +34,99 @@ impl fmt::Display for TrapReason {
             TrapReason::Message(s) => write!(f, "{}", s),
             TrapReason::I32Exit(status) => write!(f, "Exited with i32 exit status {}", status),
             TrapReason::Error(e) => write!(f, "{}", e),
+            TrapReason::InstructionTrap(code) => write!(f, "wasm trap: {}", code),
+        }
+    }
+}
+
+/// A trap code describing the reason for a trap.
+///
+/// All trap instructions have an explicit trap code.
+#[non_exhaustive]
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Hash)]
+pub enum TrapCode {
+    /// The current stack space was exhausted.
+    ///
+    /// On some platforms, a stack overflow may also be indicated by a segmentation fault from the
+    /// stack guard page.
```

Comment:
This I presume is copy-pasted, but mind dropping this clause? I believe this is no longer the case.